src/sentry/static/sentry/app/utils/tokenizeSearch.tsx

```diff
@@ -233,6 +234,65 @@ export class QueryResults {
     q.tokens = [...this.tokens];
     return q;
   }
+
+  modify(action: Actions, columnName: string, value: React.ReactText) {
+    switch (action) {
+      case Actions.ADD:
+        // If the value is null/undefined create a has !has condition.
+        if (value === null || value === undefined) {
+          // Adding a null value is the same as excluding truthy values.
+          // Remove inclusion if it exists.
+          const has = this.getTags('has');
+          if (Array.isArray(has) && has.length) {
+            this.setTag(
```

Comment:
This might be good as a new function on the QueryResults class. Maybe removeTagValue(tag, value).

Reply:
fixed